src/main/java/net/dv8tion/jda/api/interactions/commands/PrivilegeConfig.java

```diff
@@ -0,0 +1,121 @@
+import net.dv8tion.jda.api.interactions.commands.privileges.IntegrationPrivilege;
+import net.dv8tion.jda.internal.utils.Checks;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents the Privilege-Config for this integration in a Guild.
+ * <br>Can be used to get a List of {@link net.dv8tion.jda.api.interactions.commands.privileges.IntegrationPrivilege IntegrationPrivileges}
+ * applied on the application itself and on each command.
+ *
+ * @see Guild#retrieveCommandPrivileges()
+ */
+public class PrivilegeConfig
+{
+    private final Guild guild;
+    private final Map<String, List<IntegrationPrivilege>> privileges;
+
+    public PrivilegeConfig(@Nonnull Guild guild, @Nonnull Map<String, List<IntegrationPrivilege>> privileges)
+    {
+        this.guild = guild;
+        this.privileges = Collections.unmodifiableMap(privileges);
+    }
+
+    /**
+     * The guild in which this PrivilegeConfig is applied in.
+     *
+     * @return Guild in which this PrivilegeConfig is applied in.
+     */
+    @Nonnull
+    public Guild getGuild()
+    {
+        return guild;
+    }
+
+    /**
+     * The JDA-instance.
+     *
+     * @return The JDA-instance.
+     */
+    @Nonnull
+    public JDA getJDA()
+    {
+        return guild.getJDA();
+    }
+
+    /**
+     * The {@link IntegrationPrivilege IntegrationPrivileges} that have been applied to this application in this guild.
+     *
+     * <br><b>If the privileges are "Synced" (No custom config applied), this will return null.</b>
+     *
+     * <p>This does not include privileges applied to a command itself. Use {@link #getCommandPrivileges(String)} for that.
+     *
+     * @return Immutable List containing all IntegrationPrivileges that have been applied to this application in this guild.
+     */
+    @Nullable
+    public List<IntegrationPrivilege> getApplicationPrivileges()
+    {
+        return getCommandPrivileges(getJDA().getSelfUser().getApplicationId());
+    }
+
+    /**
+     * The {@link IntegrationPrivilege IntegrationPrivileges} that have been applied to the command with the given id in this guild.
+     *
+     * <br><b>If the privileges are "Synced" (No custom config applied), or a command with this id doesn't exist, this will return null.</b>
+     *
+     * <p>This does not include privileges applied to the application directly. Use {@link #getApplicationPrivileges()} for that.
+     *
+     * @param  id
+     *         The id of the command
+     *
+     * @throws IllegalArgumentException
+     *         If the provided id is null
+     *
+     * @return Immutable List containing all IntegrationPrivileges that have been applied to the command with the given id in this guild.
+     */
+    @Nullable
+    public List<IntegrationPrivilege> getCommandPrivileges(@Nonnull String id)
+    {
+        Checks.notNull(id, "Id");
+        return privileges.get(id);
+    }
```

Comment:
Maybe we should take a `Command` here that calls `getId`

Reply:
Should i add it as an overload or should i just replace it?

Reply:
An overload is better

Reply:
Yeah overload is better